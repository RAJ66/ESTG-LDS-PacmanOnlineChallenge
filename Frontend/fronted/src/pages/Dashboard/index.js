import React from "react";
import "../../Main.css";
import logo from "../../assets/logo.png";

// import { Container } from './styles';

export default function Dashboard() {
  return (
    <div className="container-main">
      <div className="header-container">
        <div className="header">
          <h1>PacmanOnlineChallenge</h1>
          <img src={logo} alt="Logo" />
        </div>
      </div>
      <div className="level-container">
        <div className="caixa">
          <h1>Level</h1>
        </div>
      </div>
      <div className="challenges-container">
        <div className="caixa">
          <h1>Challenges</h1>
        </div>
      </div>
      <div className="footer-container">
        <div className="social">
          <a href="https://github.com/RAJ66">
            <img src="https://img.icons8.com/ios-filled/50/000000/facebook-new.png" />
          </a>
          <a href="https://github.com/RAJ66">
            <img src="https://img.icons8.com/material/50/000000/instagram-new--v1.png" />
          </a>
          <a href="https://github.com/RAJ66">
            <img src="https://img.icons8.com/ios-filled/50/000000/twitter-circled.png" />
          </a>
          <a href="https://github.com/RAJ66">
            <img src="https://img.icons8.com/ios-filled/50/000000/exit.png" />
          </a>
        </div>
      </div>
    </div>
  );
}
