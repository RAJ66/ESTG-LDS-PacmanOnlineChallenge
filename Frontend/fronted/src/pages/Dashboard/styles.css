.container-dashboard {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 10vh 80vh 10vh;

  grid-template-areas:
    "header  header"
    "level  challenges"
    "footer footer";
}

.header-container {
  height: 100%;
  grid-area: header;
  background-color: var(--color-zero);
}

.header {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.stats-container {
  grid-area: level;
  background-color: var(--color-base);
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.challenges-container {
  grid-area: challenges;
  background-color: var(--color-base);
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.cashier {
  background-color: var(--color-zero);
  height: 70%;
  width: 70%;
  border-radius: var(--size-small);
}

.footer-container {
  height: 100%;
  grid-area: footer;
  background-color: var(--color-zero);

  display: flex;
  justify-content: space-around;
  align-items: center;
}

img {
  height: var(--size-medium);
  width: 46px;
}

@media only screen and (max-width: 600px) {
  .container-dashboard {
    grid-template-columns: 1fr;
    grid-template-rows: 10vh 40vh 40vh 10vh;
    grid-template-areas: "header" "level" "challenges" "footer";
  }

  body,
  input,
  button {
    font-size: 12px;
  }
}

@media only screen and (max-height: 550px) {
  img {
    height: 26px;
    width: 26px;
    margin: var(--size-small) var(--size-small) var(--size-small)
      var(--size-small);
  }
}
