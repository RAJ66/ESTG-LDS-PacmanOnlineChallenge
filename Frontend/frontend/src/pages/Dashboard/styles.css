.container-dashboard {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 10vh 80vh 10vh;

  grid-template-areas:
    "header  header"
    "level  challenges"
    "footer footer";
}

.header-container {
  height: 100%;
  grid-area: header;
  background-color: var(--color-zero);
}

.header {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.stats-container {
  grid-area: level;
  background-color: var(--color-base);
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.challenges-container {
  grid-area: challenges;
  background-color: var(--color-base);
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.cashier {
  background-color: var(--color-zero);
  height: 70%;
  width: 70%;
  border-radius: var(--size-small);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.footer-container {
  height: 100%;
  grid-area: footer;
  background-color: var(--color-zero);

  display: flex;
  justify-content: space-around;
  align-items: center;
}

img {
  height: var(--size-medium);
  width: 46px;
}

h1 {
  font-size: 40px;
  font-weight: bold;
  margin-bottom: 10px;
}

.link-button {
  background-color: transparent;
  border: none;
  cursor: pointer;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
}

.link-button:hover,
.link-button:focus {
  text-decoration: none;
}

@media only screen and (max-width: 600px) {
  .container-dashboard {
    grid-template-columns: 1fr;
    grid-template-rows: 10vh 40vh 40vh 10vh;
    grid-template-areas: "header" "level" "challenges" "footer";
    width: 100vw;
    height: 100vh;
  }

  h1 {
    font-size: 25px;
  }

  body,
  input,
  button {
    font-size: 12px;
  }
}

@media only screen and (max-height: 550px) {
  img {
    height: 26px;
    width: 26px;
    margin: var(--size-small) var(--size-small) var(--size-small)
      var(--size-small);
  }
  h1 {
    font-size: 25px;
  }
}
