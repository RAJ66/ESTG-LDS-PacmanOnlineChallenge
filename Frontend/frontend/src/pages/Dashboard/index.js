import React from "react";
import "./styles.css";
import logo from "../../assets/logo.png";

export default function Dashboard({ history }) {
  function handleClickRegister(event) {
    event.preventDefault();
    history.push("/");
  }

  return (
    <div className="container-dashboard">
      <div className="header-container">
        <div className="header">
          <h1>PacmanOnlineChallenge</h1>
          <img src={logo} alt="Logo" />
        </div>
      </div>
      <div className="stats-container">
        <div className="cashier">
          <h1>Stats</h1>
        </div>
      </div>
      <div className="challenges-container">
        <div className="cashier">
          <h1>Challenges</h1>
        </div>
      </div>
      <div className="footer-container">
        <div className="social">
          <a href="https://github.com/RAJ66">
            <img
              src="https://img.icons8.com/ios-filled/50/000000/facebook-new.png"
              alt="Facebook"
            />
          </a>
          <a href="https://github.com/RAJ66">
            <img
              src="https://img.icons8.com/material/50/000000/instagram-new--v1.png"
              alt="Instagram"
            />
          </a>
          <a href="https://github.com/RAJ66">
            <img
              src="https://img.icons8.com/ios-filled/50/000000/twitter-circled.png"
              alt="Twitter"
            />
          </a>
          <a onClick={handleClickRegister}>
            <img
              src="https://img.icons8.com/ios-filled/50/000000/exit.png"
              alt="Exit"
            />
          </a>
        </div>
      </div>
    </div>
  );
}
