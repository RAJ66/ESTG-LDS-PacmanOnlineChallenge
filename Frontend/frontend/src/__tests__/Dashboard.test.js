import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import Dashboard from "./../pages/Dashboard/index";

describe("Testing Pacman Component", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Dashboard />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
