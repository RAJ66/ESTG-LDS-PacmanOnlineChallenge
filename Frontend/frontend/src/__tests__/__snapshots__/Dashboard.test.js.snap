// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pacman Component should render correctly 1`] = `
<Dashboard>
  <div
    className="container-dashboard"
  >
    <div
      className="header-container"
    >
      <div
        className="header"
      >
        <h1>
          PacmanOnlineChallenge
        </h1>
        <img
          alt="Logo"
          src="logo.png"
        />
      </div>
    </div>
    <div
      className="stats-container"
    >
      <div
        className="cashier"
      >
        <ImageLevel>
          <div>
            <img
              alt="Level"
              className="imgLevel"
            />
          </div>
        </ImageLevel>
        <h1>
          Stats
        </h1>
        <Stats>
          <p>
            Level: 
          </p>
          <ProgressBar>
            <div
              className="progress-bar"
            >
              <div
                className="filler"
                style={
                  Object {
                    "width": "NaN%",
                  }
                }
              />
            </div>
          </ProgressBar>
          <p>
            Score: 
          </p>
          <p>
            Rank: 
          </p>
        </Stats>
      </div>
    </div>
    <div
      className="challenges-container"
    >
      <div
        className="cashier"
      >
        <h1>
          Challenges
        </h1>
        <Challenges
          nameCha1=""
          nameCha2=""
          nameCha3=""
        >
          <h3>
            <img
              alt="It a match"
              className="logo"
              src="logo.png"
            />
          </h3>
          <h3>
            <b>
               Reward:
            </b>
          </h3>
          <hr />
          <h3>
            <img
              alt="It a match"
              className="logo"
              src="logo.png"
            />
          </h3>
          <h3>
            <b>
               Reward:
            </b>
          </h3>
          <hr />
          <h3>
            <img
              alt="It a match"
              className="logo"
              src="logo.png"
            />
          </h3>
          <h3>
            <b>
               Reward:
            </b>
          </h3>
        </Challenges>
      </div>
    </div>
    <div
      className="footer-container"
    >
      <div
        className="social"
      >
        <a
          href="https://github.com/RAJ66"
        >
          <img
            alt="Facebook"
            src="https://img.icons8.com/ios-filled/50/000000/facebook-new.png"
          />
        </a>
        <a
          href="https://github.com/RAJ66"
        >
          <img
            alt="Instagram"
            src="https://img.icons8.com/material/50/000000/instagram-new--v1.png"
          />
        </a>
        <a
          href="https://github.com/RAJ66"
        >
          <img
            alt="Twitter"
            src="https://img.icons8.com/ios-filled/50/000000/twitter-circled.png"
          />
        </a>
        <button
          className="link-button "
          onClick={[Function]}
        >
          <img
            alt="Exit"
            src="https://img.icons8.com/ios-filled/50/000000/exit.png"
          />
        </button>
      </div>
    </div>
  </div>
</Dashboard>
`;
